examples/analog_out/cont_gen_voltage_wfm_int_clk_non_regen.py

```diff
@@ -1,50 +1,37 @@
 """Example of analog output voltage generation.
 
-This example demonstrates how to output a continuous periodic
-waveform with new data using an internal sample clock while
-the task is running.
+This example demonstrates how to continuously generate an
+analog output waveform by providing new data to the output buffer
+as the task is running.
+
+This example is useful if you want to generate a non-repeating waveform,
+make updates on-the-fly, or generate a frequency that is not an
+even divide-down of your sample clock. In this example,
+the default frequency value is 17.0 to demonstrate that non-regenerative output
+can be used to create a signal with a frequency that is not an even divide-down
+of your sample clock.
 """
 
-import time
-
-import numpy as np
+from analog_out_helper import create_sine_wave
 
 import nidaqmx
 from nidaqmx.constants import AcquisitionType, RegenerationMode
 
 
-def get_sine_wave_data(cycle):
-    """Generate a sine wave data."""
-    frequency = (10, 50)
-    amplitude = (1, 3)
-    sampling_rate = (100, 500)
-    duration = 1
-    selection = cycle % 2
-
-    # generate the time array
-    t = np.arange(0, duration, 1 / sampling_rate[selection])
-    # Generate the sine wave
-    data = amplitude[selection] * np.sin(2 * np.pi * frequency[selection] * t)
-    return data
-
-
 with nidaqmx.Task() as task:
     is_first_run = True
+    sampling_rate = 1000.0
     task.ao_channels.add_ao_voltage_chan("Dev1/ao0")
     task.out_stream.regen_mode = RegenerationMode.DONT_ALLOW_REGENERATION
-    task.timing.cfg_samp_clk_timing(
-        1000, sample_mode=AcquisitionType.CONTINUOUS, samps_per_chan=100
-    )
+    task.timing.cfg_samp_clk_timing(sampling_rate, sample_mode=AcquisitionType.CONTINUOUS)
 
     try:
-        cycle = 1
-        print("Starting task. Press Ctrl+C to stop.")
-        time.sleep(1.0)
-
+        print("Generating voltage continuously. Press Ctrl+C to stop.")
         while True:
-            data = get_sine_wave_data(cycle)
+            data = create_sine_wave(
+                frequency=17.0, amplitude=1.0, sampling_rate=sampling_rate, duration=1.0
+            )
             task.write(data)
```

Comment:
This doesn't demonstrate a non-repeating waveform. Each loop iteration generates the same waveform chunk, and the number of samples in the chunk is based on the sample rate.

Look at the corresponding LabVIEW example. 
![image](https://github.com/ni/nidaqmx-python/assets/17914399/3d616887-ec85-4a13-86e2-b039f42a0d88)

It doesn't generate the same sine wave data for each loop iteration. Instead, each iteration generates a new sine wave chunk that lines up with the previous sine wave chunk. The chunks do not necessarily contain an integral number of sine wave cycles. This is how you generate a continuous sine wave at a frequency that is not related to your sample rate and buffer size.

To demonstrate this, the `create_sine_wave` function should also take the number of samples to generate and the starting phase, and return the ending phase to use for the next call.

Reply:
[numpy.linspace](https://numpy.org/doc/stable/reference/generated/numpy.linspace.html#numpy.linspace) might be helpful. Or maybe it's easier to build a list with range() and use the `math` module.

Reply:
Two examples to clarify:

1. The frequency you are generating is so slow, you can only generate one quadrant of the sine wave at a time.
- 1st iteration generates time range [0, 0.5*pi)
- 2nd iteration generates time range [0.5*pi, 1.0*pi)
- 3rd iteration generates time range [1.0*pi, 1.5*pi)
- 4th iteration generates time range [1.5*pi, 2*pi)
- 5th iteration generates time range [2*pi, 2.5*pi)

For each chunk, the phase input is the time offset at the start of the time range, and the phase output is the time offset at the end of the time range. Note that the time ranges are partially open intervals; we don't want to repeat the last sample of chunk N as the 1st sample of chunk N+1.

The sine function is cyclical and its implementation in computers gets less accurate as you get farther from the Y-axis, so it would make sense to keep the phase within 1 cycle of the Y-axis using the `%` operator or something. In this example, the 4th iteration would return an ending phase of 0 instead of 2*pi. 

Note that for many frequencies, the phase may never go back to exactly 0 at the end of a buffer. Or it may take a large number of iterations to do so. That's expected. In these situations, generating repeated data would introduce frequency error, and generating non-repeating data corrects that frequency error.

2. The frequency you are generating may translate to multiple sine wave cycles, then a fractional one. For example, one full cycle plus one quadrant.
- 1st iteration generates time range [0, 2.5*pi)
- 2nd iteration generates time range [2.5*pi, 5.0*pi)
- 3rd iteration generates time range [5.0*pi, 7.5*pi)
- 4th iteration generates time range [7.5*pi, 10.0*pi)

Keeping the phase within 2*pi by modding:
- 1st iteration phase in: 0, generate [0, 2.5*pi), phase out: 2.5*pi % 2*pi -> 0.5*pi
- 2nd iteration phase in: 0.5*pi, generate [0.5*pi, 3.0*pi), phase out: 3.0*pi % 2*pi -> 1.0*pi
- 3rd iteration phase in: 1.0*pi, generate [1.0*pi, 3.5*pi), phase out: 3.5*pi % 2*pi -> 1.5*pi
- 4th iteration phase in: 1.5*pi, generate [1.5*pi, 4.0*pi), phase out: 4.0*pi % 2*pi -> 0.0

Reply:
'''
The frequency you are generating may translate to multiple sine wave cycles, then a fractional one. For example, one full cycle plus one quadrant.
1st iteration generates time range [0, 2.5*pi)
2nd iteration generates time range [2.5*pi, 5.0*pi)
3rd iteration generates time range [5.0*pi, 7.5*pi)
4th iteration generates time range [7.5*pi, 10.0*pi)
Keeping the phase within 2*pi by modding:

1st iteration phase in: 0, generate [0, 2.5*pi), phase out: 2.5*pi % 2*pi -> 0.5*pi
2nd iteration phase in: 0.5*pi, generate [0.5*pi, 3.0*pi), phase out: 3.0*pi % 2*pi -> 1.0*pi
3rd iteration phase in: 1.0*pi, generate [1.0*pi, 3.5*pi), phase out: 3.5*pi % 2*pi -> 1.5*pi
4th iteration phase in: 1.5*pi, generate [1.5*pi, 4.0*pi), phase out: 4.0*pi % 2*pi -> 0.0
'''

I made changes to the comment above, hopefully, I get the points correctly. 
I tested with physical device, so far, I can see the waveform on the scope.